Ginger/Ginger/ALM/ALMIntegration.cs

```diff
@@ -65,51 +62,106 @@ public void UpdateALMType(eALMType AlmType)
-            AlmCore.SetALMConfigurations(WorkSpace.Instance.Solution.ALMServerURL, WorkSpace.Instance.Solution.UseRest, WorkSpace.Instance.UserProfile.ALMUserName, WorkSpace.Instance.UserProfile.ALMPassword, WorkSpace.Instance.Solution.ALMDomain, WorkSpace.Instance.Solution.ALMProject, WorkSpace.Instance.Solution.ALMProjectKey);
-            SyncConfigurations();
+            if(CurrentAlmConfigurations != null)
+                AlmCore.SetALMConfigurations(CurrentAlmConfigurations.ALMServerURL, CurrentAlmConfigurations.UseRest, CurrentAlmConfigurations.ALMUserName, CurrentAlmConfigurations.ALMPassword, CurrentAlmConfigurations.ALMDomain, CurrentAlmConfigurations.ALMProjectName, CurrentAlmConfigurations.ALMProjectKey, CurrentAlmConfigurations.AlmType, CurrentAlmConfigurations.ALMConfigPackageFolderPath);
         }
 
         public void SyncConfigurations()
         {
-            WorkSpace.Instance.Solution.ALMServerURL = ALMCore.AlmConfig.ALMServerURL;
-            WorkSpace.Instance.Solution.UseRest = ALMCore.AlmConfig.UseRest;
-            WorkSpace.Instance.UserProfile.ALMUserName = ALMCore.AlmConfig.ALMUserName;
-            WorkSpace.Instance.UserProfile.ALMPassword = ALMCore.AlmConfig.ALMPassword;
-            WorkSpace.Instance.Solution.ALMDomain = ALMCore.AlmConfig.ALMDomain;
-            WorkSpace.Instance.Solution.ALMProject = ALMCore.AlmConfig.ALMProjectName;
-            WorkSpace.Instance.Solution.ALMProjectKey = ALMCore.AlmConfig.ALMProjectKey;
-            if ((eALMType)WorkSpace.Instance.Solution.AlmType == eALMType.Jira)
+            
+        }
+
+        public GingerCoreNET.ALMLib.ALMConfig GetDefaultAlmConfig()
+        {
+            GingerCoreNET.ALMLib.ALMConfig AlmConfig = WorkSpace.Instance.Solution.ALMConfigs.Where(x => x.DefaultAlm == true).FirstOrDefault();
+            if (AlmConfig == null)
             {
-                WorkSpace.Instance.Solution.ConfigPackageFolderPath = ALMCore.AlmConfig.ALMConfigPackageFolderPath;
+                AlmConfig = new GingerCoreNET.ALMLib.ALMConfig();
+                AlmConfig.DefaultAlm = true;
+                WorkSpace.Instance.Solution.ALMConfigs.Add(AlmConfig);
             }
+            return AlmConfig;
         }
+        public bool SetDefaultAlmConfig(GingerCoreNET.ALMLib.ALMIntegration.eALMType AlmType)
+        {
+            //set default on the solution
+            foreach (GingerCoreNET.ALMLib.ALMConfig alm in WorkSpace.Instance.Solution.ALMConfigs.Where(x => x.DefaultAlm == true).ToList())
+            {
+                alm.DefaultAlm = false;
+            }
+            GingerCoreNET.ALMLib.ALMConfig almConfig = GetCurrentAlmConfig(AlmType);
+            almConfig.DefaultAlm = true;
 
-        public ALMConfig AlmConfigurations
+            //set default on almcore
+            foreach (GingerCoreNET.ALMLib.ALMConfig alm in ALMCore.AlmConfigs.Where(x => x.DefaultAlm == true).ToList())
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)